### ISSUE TITLE
Don't only look for variable definitions in parent & sibling nodes

### ISSUE THREAD

**Issue by felixfbecker** [felixfbecker/php-language-server#160]
```php
if ($whatever) {
  $x = 1;
} else {
  $x = 2;
}
echo $x;
```
It's not possible to get the definition of `$x` because it is not defined in a parent or sibling node, but in a child node of a sibling. Functions must be the only boundary to variable scopes.

**Comment by jens1o** [comment]
I think it is possible, but it wouldn't be efficient, is it?

**Comment by felixfbecker** [comment]
I don't think it would be too bad given it's still restricted by the function boundaries.

**Comment by jens1o** [comment]
I'll take a look, but I'm not sure wether I can do that.

**Comment by jens1o** [comment]
That actually worked without even a single line being added:
![image](https://cloud.githubusercontent.com/assets/11234139/24867238/df1744c0-1e0c-11e7-9766-d9a053b0eccd.png)

**Comment by felixfbecker** [comment]
That's completion, try to get the definition

**Comment by jens1o** [comment]
Ah, and what do you expect to have? The first one or the second?

**Comment by felixfbecker** [comment]
I wouldn't care. Just not nothing.

**Comment by jens1o** [comment]
Okay, I'll look. I think that may be a good exercise for me.
